Bugfix/4035

OlliTietavainenVaadin opened the pull request:
Fixes issue #4035 Flash component doesn't resize correctly. Now if Flash component is set to 100% (or other percentage size), it fills up the slot correctly and it also resizes without reloading if the window or parent component is resized.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/8552)
<!-- Reviewable:end -->

OlliTietavainenVaadin committed:
fixes issue #4035: allows Flash embed to expand if percentage dimensions given

OlliTietavainenVaadin committed:
Merge branch '7.7' of https://github.com/vaadin/framework into bugfix/4035

OlliTietavainenVaadin committed:
removed unused method

OlliTietavainenVaadin committed:
add ticket number to FlashExpansion test class

tsuoanttila commented on uitest/src/main/java/com/vaadin/tests/components/flash/FlashExpansion.java:
```diff
@@ -0,0 +1,42 @@
+package com.vaadin.tests.components.flash;
```
Missing license header from this file.

OlliTietavainenVaadin committed:
added license header to FlashExpansion.java

OlliTietavainenVaadin commented:
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[uitest/src/main/java/com/vaadin/tests/components/flash/FlashExpansion.java, line 1 at r1](https://reviewable.io:443/reviews/vaadin/framework/8552#-Kcrln5qZpkfqwjPBb4z-r1-1:-KcwDTAX6AUclXv_zVmm:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/669f328fe89dcdb3d5a8a71147f2f1dbf85e065b/uitest/src/main/java/com/vaadin/tests/components/flash/FlashExpansion.java#L1)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

Missing license header from this file.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8552)*
<!-- Sent from Reviewable.io -->

OlliTietavainenVaadin committed:
License needs to stop at 2014 or validation won't pass

pleku commented:
Reviewed 1 of 3 files at r1, 1 of 2 files at r2.
Review status: 2 of 3 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/8552#-KemeTc8vKWF3zL5LKKR:-KemeTc8vKWF3zL5LKKS:b-ypsfr6) (no related file):*
Instead of adding more code about parsing relative sizing, please reuse `com.vaadin.client.Util` methods like `parseRelativeSizing(AbstractComponentState)`, which internally also uses `WidgetUtil.parseRelativeSize(String)`

---

*[client/src/main/java/com/vaadin/client/ui/VFlash.java, line 138 at r2](https://reviewable.io:443/reviews/vaadin/framework/8552#-Kemd-DREY0rDue0KDkc:-Kemd-DREY0rDue0KDkd:b-n7ba8t) ([raw file](https://github.com/vaadin/framework/blob/0a8cf4405b97241add31d0840f78841563958b15/client/src/main/java/com/vaadin/client/ui/VFlash.java#L138)):*
> ```Java
>     }
> 
>     public void setSlotHeightAndWidth(int slotOffsetHeight,
> ```

Please add javadocs on public methods when adding new code to maintain. I know it is not the case in the other methods here, but let's not

---

*[client/src/main/java/com/vaadin/client/ui/VFlash.java, line 272 at r2](https://reviewable.io:443/reviews/vaadin/framework/8552#-KemdprsC0fh8lzgF4xI:-KemdprsC0fh8lzgF4xJ:b-vkpxz9) ([raw file](https://github.com/vaadin/framework/blob/0a8cf4405b97241add31d0840f78841563958b15/client/src/main/java/com/vaadin/client/ui/VFlash.java#L272)):*
> ```Java
>     }
> 
>     protected void resizeEmbedElement() {
> ```

why is this protected ? would need javadocs then but I'd just change it to private

---

*[client/src/main/java/com/vaadin/client/ui/VFlash.java, line 294 at r2](https://reviewable.io:443/reviews/vaadin/framework/8552#-Kemdzm5KSGZ1fby3Jk8:-Kemdzm66A4oJC3UiTiZ:b-vkpxz9) ([raw file](https://github.com/vaadin/framework/blob/0a8cf4405b97241add31d0840f78841563958b15/client/src/main/java/com/vaadin/client/ui/VFlash.java#L294)):*
> ```Java
>     }
> 
>     protected String getRelativePixelWidth() {
> ```

why is this protected ? would need javadocs then but I'd just change it to private

---

*[uitest/src/main/java/com/vaadin/tests/components/flash/FlashExpansion.java, line 24 at r2](https://reviewable.io:443/reviews/vaadin/framework/8552#-Kemf14VvmOwFvQmJDc8:-Kemf14VvmOwFvQmJDc9:b4f4pi0) ([raw file](https://github.com/vaadin/framework/blob/0a8cf4405b97241add31d0840f78841563958b15/uitest/src/main/java/com/vaadin/tests/components/flash/FlashExpansion.java#L24)):*
> ```Java
> import com.vaadin.ui.Flash;
> 
> public class FlashExpansion extends TestBase {
> ```

Why is there not an automated test for verifying this ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8552)*
<!-- Sent from Reviewable.io -->

pleku commented:
Reviewed 1 of 2 files at r2.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8552)*
<!-- Sent from Reviewable.io -->

hesara commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[client/src/main/java/com/vaadin/client/ui/VFlash.java, line 106 at r2](https://reviewable.io:443/reviews/vaadin/framework/8552#-KemQY10NnO1xfG80ccU:-KemQY10NnO1xfG80ccV:bjfey65) ([raw file](https://github.com/vaadin/framework/blob/df1750f19c059ca94106a828fe40cb1c08c9923f/client/src/main/java/com/vaadin/client/ui/VFlash.java#L106)):*
> ```Java
>     @Override
>     public void setWidth(String width) {
>         // super.setWidth(height);
> ```

(I guess these comments were here to indicate that we explicitly don't want to call super, although a more descriptive comment would be better.)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8552)*
<!-- Sent from Reviewable.io -->

OlliTietavainenVaadin committed:
added UI test for Flash expansion

OlliTietavainenVaadin committed:
added comments, protected -> private and use of WidgetUtil instead of custom calc code

OlliTietavainenVaadin committed:
Merge branch '7.7' of https://github.com/vaadin/framework into bugfix/4035

OlliTietavainenVaadin commented:
Review status: 2 of 4 files reviewed at latest revision, 7 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/8552#-KemeTc8vKWF3zL5LKKR:-Kff-5U8DSVPTCwVE0Kp:bb6un1j) (no related file):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Instead of adding more code about parsing relative sizing, please reuse `com.vaadin.client.Util` methods like `parseRelativeSizing(AbstractComponentState)`, which internally also uses `WidgetUtil.parseRelativeSize(String)`
</blockquote></details>

Done

---

*[client/src/main/java/com/vaadin/client/ui/VFlash.java, line 106 at r2](https://reviewable.io:443/reviews/vaadin/framework/8552#-KemQY10NnO1xfG80ccU:-Kff-9l0sAi30r4_mtiC:b-9jfcox) ([raw file](https://github.com/vaadin/framework/blob/df1750f19c059ca94106a828fe40cb1c08c9923f/client/src/main/java/com/vaadin/client/ui/VFlash.java#L106)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(I guess these comments were here to indicate that we explicitly don't want to call super, although a more descriptive comment would be better.)
</blockquote></details>

Added better comments

---

*[client/src/main/java/com/vaadin/client/ui/VFlash.java, line 138 at r2](https://reviewable.io:443/reviews/vaadin/framework/8552#-Kemd-DREY0rDue0KDkc:-Kff-Brrbcnl4uxHwsHJ:b-b6dyce) ([raw file](https://github.com/vaadin/framework/blob/0a8cf4405b97241add31d0840f78841563958b15/client/src/main/java/com/vaadin/client/ui/VFlash.java#L138)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Please add javadocs on public methods when adding new code to maintain. I know it is not the case in the other methods here, but let's not
</blockquote></details>

Added javadocs

---

*[client/src/main/java/com/vaadin/client/ui/VFlash.java, line 272 at r2](https://reviewable.io:443/reviews/vaadin/framework/8552#-KemdprsC0fh8lzgF4xI:-Kff-Dpxo1RBzcEklYZu:b-fzvzgx) ([raw file](https://github.com/vaadin/framework/blob/0a8cf4405b97241add31d0840f78841563958b15/client/src/main/java/com/vaadin/client/ui/VFlash.java#L272)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

why is this protected ? would need javadocs then but I'd just change it to private
</blockquote></details>

Changed to private

---

*[client/src/main/java/com/vaadin/client/ui/VFlash.java, line 294 at r2](https://reviewable.io:443/reviews/vaadin/framework/8552#-Kemdzm5KSGZ1fby3Jk8:-Kff-FN_UEuXrcogqfIy:b-fzvzgx) ([raw file](https://github.com/vaadin/framework/blob/0a8cf4405b97241add31d0840f78841563958b15/client/src/main/java/com/vaadin/client/ui/VFlash.java#L294)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

why is this protected ? would need javadocs then but I'd just change it to private
</blockquote></details>

Changed to private

---

*[uitest/src/main/java/com/vaadin/tests/components/flash/FlashExpansion.java, line 24 at r2](https://reviewable.io:443/reviews/vaadin/framework/8552#-Kemf14VvmOwFvQmJDc8:-Kff-GnIDAhwsZzMnQVE:bl3dunu) ([raw file](https://github.com/vaadin/framework/blob/0a8cf4405b97241add31d0840f78841563958b15/uitest/src/main/java/com/vaadin/tests/components/flash/FlashExpansion.java#L24)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Why is there not an automated test for verifying this ?
</blockquote></details>

Added FlashExpansionTest. It's not very easy writing a test at all, since you can only test Flash with IE.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8552)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[uitest/src/test/java/com/vaadin/tests/components/flash/FlashExpansionTest.java, line 21 at r3](https://reviewable.io:443/reviews/vaadin/framework/8552#-Kffo2Qq-FjHQbKx0A1X:-Kffo2Qq-FjHQbKx0A1Y:b8svjrz) ([raw file](https://github.com/vaadin/framework/blob/67212002b0697b90f7bf4b4543b814994aeca39f/uitest/src/test/java/com/vaadin/tests/components/flash/FlashExpansionTest.java#L21)):*
> ```Java
>     public List<DesiredCapabilities> getBrowsersToTest() {
>         // No Flash support in Chrome, FF, PhantomJS
>         List<DesiredCapabilities> capabilities = getBrowsersSupportingShiftClick();
> ```

not really related to shift click, so probably should create the list of supported browsers (IE?) here directly with getBrowserCapabilities(Browser.IE...) or such

---

*[uitest/src/test/java/com/vaadin/tests/components/flash/FlashExpansionTest.java, line 32 at r3](https://reviewable.io:443/reviews/vaadin/framework/8552#-KffoMj-IQsrJJR5hz52:-KffoMj-IQsrJJR5hz53:b-3m60wc) ([raw file](https://github.com/vaadin/framework/blob/67212002b0697b90f7bf4b4543b814994aeca39f/uitest/src/test/java/com/vaadin/tests/components/flash/FlashExpansionTest.java#L32)):*
> ```Java
>         openTestURL();
>         /* Allow the flash plugin to load */
>         sleep(5000);
> ```

sleep is quite ugly here - is there some (DOM?) condition that we could wait for instead?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8552)*
<!-- Sent from Reviewable.io -->

OlliTietavainenVaadin committed:
added method to get Flash supporting browsers, removed sleep(5000)

OlliTietavainenVaadin commented:
Review status: 3 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

*[uitest/src/test/java/com/vaadin/tests/components/flash/FlashExpansionTest.java, line 21 at r3](https://reviewable.io:443/reviews/vaadin/framework/8552#-Kffo2Qq-FjHQbKx0A1X:-KfjafjZZ1GjIz4S1rFX:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/67212002b0697b90f7bf4b4543b814994aeca39f/uitest/src/test/java/com/vaadin/tests/components/flash/FlashExpansionTest.java#L21)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

not really related to shift click, so probably should create the list of supported browsers (IE?) here directly with getBrowserCapabilities(Browser.IE...) or such
</blockquote></details>

Done.

---

*[uitest/src/test/java/com/vaadin/tests/components/flash/FlashExpansionTest.java, line 32 at r3](https://reviewable.io:443/reviews/vaadin/framework/8552#-KffoMj-IQsrJJR5hz52:-KfjagbI9FSZWDfNIAEq:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/67212002b0697b90f7bf4b4543b814994aeca39f/uitest/src/test/java/com/vaadin/tests/components/flash/FlashExpansionTest.java#L32)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

sleep is quite ugly here - is there some (DOM?) condition that we could wait for instead?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8552)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8552)*
<!-- Sent from Reviewable.io -->
